src/DepFile.cpp

```diff
@@ -69,51 +184,71 @@ static bool sParseMakeDepFile(FileID inDepFileID, StringView inDepFileContent, s
 	}
 
 	dep_file_content = dep_file_content.substr(deps_start + cDepStartMarker.size());
-
+	
 	while (!dep_file_content.empty())
 	{
 		// Skip white space before the path.
 		while (!dep_file_content.empty() && sIsSpace(dep_file_content.front()))
 			dep_file_content = dep_file_content.substr(1);
 
 		// Lines generally end with space + backslash + linefeed.
+		constexpr StringView cWindowsLineEnd  = " \\\r\n";
 		constexpr StringView cLineEnd  = " \\\n";
-		size_t path_end  = dep_file_content.find(cLineEnd);
-		size_t next_path = path_end + cLineEnd.size();
+		size_t path_end  = dep_file_content.find(cWindowsLineEnd);
+		size_t next_path = path_end + cWindowsLineEnd.size();
+
+		// Match first on Windows' CRLF before matching on LF.
+		if (path_end == StringView::npos)
+		{
+			path_end  = dep_file_content.find(cLineEnd);
+			next_path = path_end + cLineEnd.size();
+		}
 		if (path_end == StringView::npos)
 		{
 			// Last line might also end with a linefeed (or nothing).
-			constexpr StringView cLastLineEnd = "\n";
-			path_end  = dep_file_content.find(cLastLineEnd);
+			constexpr StringView cLastLineEnd = "\n\r";
+			path_end = dep_file_content.find_first_of(cLastLineEnd);
 			next_path = path_end + cLastLineEnd.size();
```

Comment:
There is still one issue here `find_first_of(cLastLineEnd)` matched only 1 character, so it should be +1 rather than +size().
Not sure if it's possible to have a Windows end line with both characters here? With DXC it's only \n it seems

Reply:
Nice catch, thanks!
> Not sure if it's possible to have a Windows end line with both characters here?

Glslang wrote CRLF at the end of my dependency files. I did a little big of digging and of [ea087ff](https://github.com/KhronosGroup/glslang/blob/ea087ff90d03947307cfe52500b74551aa35d34d/StandAlone/StandAlone.cpp#L1283), it just opens the file as a default `std::ofstream` and writes a final `std::endl`, I think it's safe to assume there'll be terminal CRLF in at least some of those files.

Reply:
But then doesn't that mean that `next_path` points to the `\n` instead of after it? (because find_first_of found the \r first)

Then it probably does another iteration and it's not very clear what happens